pkg/auth/dependency/interaction/provider_commit.go

```diff
@@ -206,6 +208,103 @@ func (p *Provider) onCommitRemoveIdentity(i *Interaction, intent *IntentRemoveId
 	return nil
 }
 
+func (p *Provider) onCommitUpdateIdentity(i *Interaction, intent *IntentUpdateIdentity, userID string) error {
+	err := p.checkIdentitiesDuplicated(i.UpdateIdentities)
+	if err != nil {
+		return err
+	}
+
+	if len(i.UpdateIdentities) != 1 {
+		panic("interaction: unexpected number of identities to be updated")
+	}
+
+	var originalIdentityInfo *IdentityInfo
+	updateIdentityInfo := i.UpdateIdentities[0]
+
+	// check if there is any authenticators need to be deleted after identity update
+	keepAuthenticators := map[string]*AuthenticatorInfo{}
+	removeAuthenticators := map[string]*AuthenticatorInfo{}
```

Comment:
I think it can be named `allAuthenticators` for more accurate name, since we want to remove all authenticators that does not need to be kept.

Reply:
@kiootic Not sure what you mean, change name of `keepAuthenticators` or `removeAuthenticators`?

Currently `keepAuthenticators` are authenticators that we want to keep, `removeAuthenticators` are authenticators that related to the outdated identity. 

And now we remove authenticators that inside `removeAuthenticators` but not `keepAuthenticators`.

Reply:
`removeAuthenticators` -> `allAuthenticators`.

We want to remove all authenticators of a user (`allAuthenticators`), except those we explicitly want to keep (`keepAuthenticators`).

Reply:
But `removeAuthenticators` is not containing all authenticators, it only contains authenticators that related to the outdated identity. 🤔

Reply:
Oh I see, then let's just keep the current name. It just confused me a bit when I realized that some authenticators in `removeAuthenticators` may not be removed.

Reply:
Updated a bit in 8d479e7, see if it looks less confusing?:)